File: 백준/Silver/11723. 집합/집합.py
import sys

m = int(input())
S = set()

for _ in range(m):
    # comm = list(input().split())
    comm = sys.stdin.readline().strip().split()
    
    if len(comm) == 1:
        if comm[0] == 'all': # all
            S = set([i for i in range(1, 21)])
        else: # empty
            S = set()
    else:
        func, x = comm[0], int(comm[1])
        if func == 'add':
            S.add(x)
        elif func == 'remove':
            S.discard(x) # 삭제할 요소가 없으면 무시
            # remove()는 에러 발생
        elif func == 'check':
            if x in S:
                print(1)
            else:
                print(0)
        elif func == 'toggle':
            if x in S:
                S.discard(x)
            else:
                S.add(x)

File: 백준/Silver/11723. 집합/집합.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.HashSet;
import java.util.Set;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();
        int M = Integer.parseInt(br.readLine());
        Set<Integer> s = new HashSet<>();
        while (M-- > 0) {
            st = new StringTokenizer(br.readLine());
            String command = st.nextToken();
            if (command.equals("add")) {
                int x = Integer.parseInt(st.nextToken());
                s.add(x);
            } else if (command.equals("remove")) {
                int x = Integer.parseInt(st.nextToken());
                s.remove(x);
            } else if (command.equals("check")) {
                int x = Integer.parseInt(st.nextToken());
                if (s.contains(x)) sb.append("1\n");
                else sb.append("0\n");

            } else if (command.equals("toggle")) {
                int x = Integer.parseInt(st.nextToken());
                if (s.contains(x)) s.remove(x);
                else s.add(x);
            } else if (command.equals("all")) {
                s.clear();
                for (int i = 1; i <= 20; i++) s.add(i);
            } else if (command.equals("empty")) {
                s.clear();
            }
        }
        System.out.print(sb);
    }
}